src/main/java/com/aws/greengrass/util/platforms/windows/WindowsExec.java

```diff
@@ -80,6 +70,13 @@ public Path which(String fn) {
     @Override
     public String[] getCommand() {
         String[] decorated = Arrays.copyOf(cmds, cmds.length);
+        for (int i = 0; i < decorated.length; i++) {
+            final String arg = decorated[i];
+            // Space and \t require quoting otherwise won't be treated as a single arg
+            if (!isQuoted(arg) && arg.matches(".*[ \t].*")) {
+                decorated[i] = "\"" + arg + "\"";
+            }
+        }
```

Comment:
Where is this logic from, if anywhere? You had quoting logic before from Java didn't you?

Reply:
Also, this logic applies when not using runas as well, right? Is it equally applicable?

Reply:
Space and tab is the most basic check from Windows [ProcessImpl](https://github.com/openjdk/jdk/blob/f9b2507f3e86bcb91e8ccfd0a84f31712fd535c2/src/java.base/windows/classes/java/lang/ProcessImpl.java#L226). So this is essentially what JDK does by default.

When not using runas, ProcessBuilder won't double quote the args according to [needsEscaping](https://github.com/openjdk/jdk/blob/f9b2507f3e86bcb91e8ccfd0a84f31712fd535c2/src/java.base/windows/classes/java/lang/ProcessImpl.java#L302).